python-checks/src/test/resources/checks/confusingTypeChecking/nonCallableCalled.py

```diff
@@ -0,0 +1,21 @@
+from typing import Set, FrozenSet
+
+def foo(param1: int, param2: Set[int], param3: FrozenSet[int]):
+  param1() # Noncompliant {{Fix this call; Previous type checks suggest that "param1" has type int and it is not callable.}}
+  x = 42
+  x() # OK, raised by S5756
+
+  param2() # Noncompliant
+  s = set()
+  s() # OK, raised by S5756
+
+  param3() # Noncompliant
+  fs = frozenset()
+  fs() # OK, raised by S5756
+
+def derived(x: int):
+  x.conjugate()() # FN
+  y = x or 'str'
+  y() # Noncompliant
+  z = x + 'str'
+  z() # FN?
```

Comment:
Not sure we should consider this a FN. We should raise a code smell for the impossible `+` operation above, though, then we shouldn't be able to resolve a type for `z` I assume.
I believe this one would be a FN though (we should infer a DeclaredType of `str` for `y`):
```
def foo(x: str):
  y = x + "hello"
  y() # FN
```

Reply:
yes indeed, I agree